docs/docs/how-gatsby-works-with-github-pages.md

```diff
@@ -77,3 +77,56 @@ After running `npm run deploy` you should see your website at `username.github.i
 If you use a [custom domain](https://help.github.com/articles/using-a-custom-domain-with-github-pages/), don't add a `pathPrefix` as it will break navigation on your site. Path prefixing is only necessary when the site is _not_ at the root of the domain like with repository sites.
 
 **Note**: Don't forget to add your [CNAME](https://help.github.com/articles/troubleshooting-custom-domains/#github-repository-setup-errors) file to the `static` directory.
+
+### Deploying to GitHub pages from a CI server
+
+It's also possible to deploy your website to `gh-pages` through a CI server. This example uses Travis CI, a hosted Continuous Integration service, but other CI systems could work as well.
+
+You can use the [gh-pages npm module](https://www.npmjs.com/package/gh-pages) to deploy. But first, you need to configure it with proper credentials so that `gh-pages` is able to push a new branch.
+
+#### Obtain a GitHub token for authenticating with CI
+
+To push changes from the CI system to GitHub, you'll need to authenticate. It's recommended to use [GitHub developer tokens](https://help.github.com/en/articles/creating-a-personal-access-token-for-the-command-line).
+
+In GitHub go to your account settings -> Developer settings -> Personal access tokens, and create a new token that provides the `repo` access permissions.
+
+In Travis's configuration for the repository, add a new secret environment variable of the name `GH_TOKEN` with the value of the token copied from GitHub. Make sure you **DO NOT toggle the "display in build logs" setting to on** as the token should remain secret. Otherwise, strangers would be able to push to your repository (a big security issue).
+
+#### Step 2 - deploy run script
```

Comment:
```suggestion
#### Add script to deploy to GitHub Pages via CI
```
Like above, these headings shouldn't have steps in them since this isn't a guided tutorial. Making the titles descriptive should be sufficient.